### ISSUE TITLE
Postgresql optimization: Slow query on carddav_addressbooks

### ISSUE THREAD

**Issue by kearva** [blind-coder/rcmcarddav#291]
We noticed a large CPU usage on our database, which were traced to sequential scans on carddav_addressbooks.
```sql
SELECT id FROM carddav_addressbooks WHERE user_id='xxxx' AND active=1
```

I did an **EXPLAIN** on this query and got:
```
                                      QUERY PLAN                                      
--------------------------------------------------------------------------------------
 Gather  (cost=1000.00..6575.49 rows=1 width=4)
   Workers Planned: 2
   ->  Parallel Seq Scan on carddav_addressbooks  (cost=0.00..5575.39 rows=1 width=4)
         Filter: ((user_id = xxxx) AND (active = 1))
```
I suggest to include this index in the setup for postgres:
```sql
CREATE INDEX carddav_addressbooks_user_id on carddav_addressbooks (user_id);
```

After I included this index, the CPU usage went from 100 % to about 10 %. I did another **EXPLAIN** and the cost says it all:
```
                                               QUERY PLAN                                                
---------------------------------------------------------------------------------------------------------
 Index Scan using carddav_addressbooks_user_id on carddav_addressbooks  (cost=0.42..8.44 rows=1 width=4)
   Index Cond: (user_id = xxxx)
   Filter: (active = 1)
(3 rows)
````

**Comment by mstilkerich** [comment]
Thank you for this nicely researched issue report. I will add the index on the v4 development branch. It would be even better to avoid the database query that happens on pretty much every page load, just for the purpose to provide roundcube with a list of carddav addressbooks to query for auto completion. But that would require a lot more work, so for now I guess the index is the way to go for.

**Comment by mstilkerich** [comment]
This is fixed by commit ed36d7a1aad3f756149904b6a5101e80a14cbe3d. It also adds indexes on other foreign key columns that are frequently used in lookups. If you find the issue persists, please comment / reopen the issue.
